ADAL/tests/ADAcquireTokenTests.m

```diff
@@ -1622,4 +1623,89 @@ - (void)testAllowSilentRequestParameters
     TEST_WAIT;
 }
 
+- (void)testSkipCacheRequestParameters_whenSkipCacheIsSet_shouldSkipCache
+{
+    ADAuthenticationError* error = nil;
+    ADAuthenticationContext *context = [self getTestAuthenticationContext];
+    ADRequestParameters *params = [[ADRequestParameters alloc] initWithAuthority:context.authority
+                                                                        resource:TEST_RESOURCE
+                                                                        clientId:TEST_CLIENT_ID
+                                                                     redirectUri:TEST_REDIRECT_URL.absoluteString
+                                                                      identifier:[ADUserIdentifier identifierWithId:TEST_USER_ID]
+                                                                      tokenCache:context.tokenCacheStore
+                                                                extendedLifetime:NO
+                                                                   correlationId:nil
+                                                              telemetryRequestId:nil];
+    
+    // Add a token item to return in the cache
+    ADTokenCacheItem* item = [self adCreateCacheItem];
+    [context.tokenCacheStore.dataSource addOrUpdateItem:item correlationId:nil error:&error];
+    
+    // No skipCache is set, cached item should be found
+    ADAuthenticationRequest *req = [ADAuthenticationRequest requestWithContext:context requestParams:params error:nil];
+    [req acquireToken:@"123"
+      completionBlock:^(ADAuthenticationResult *result)
+     {
+         XCTAssertEqual(result.status, AD_SUCCEEDED);
+         XCTAssertNotNil(result.tokenCacheItem);
+         
+         TEST_SIGNAL;
+     }];
+    
+    TEST_WAIT_NOT_BLOCKING_MAIN_QUEUE;
+    
```

Comment:
I'd suggest splitting this test into 2:
1. No skipCache is set, cached item should be found
2. skipCache is set, cache should be skipped and web view controller should be hit

Because nr 1 does/tests not what the test method's name is telling.